src/types.ts

```diff
@@ -0,0 +1,13 @@
+import {z} from 'zod'
+
+export const appDataSchema = z.object({
+  schema: z.string(),
+  values: z.array(z.string()),
+})
```

Comment:
zod within zod within zod 

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExZGF3aGZnamhwN2xldGFtZXJqbjhuMjQ1ajM5M2kxeTA3eHJnbGJjcSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/26ufdipQqU2lhNA4g/giphy.gif)

Reply:
Inception 🤯

Reply:
This is redundant, decompression or parse would fail if the url is not valid. Would remove it.

Reply:
I disagree. While decompression and JSON.parse can handle the transformation of the data, they do not validate the structure of the data. The schema ensures at runtime that the data coming from the URL matches the expected structure. This is crucial as it allows us to catch any inconsistencies or errors in the data early on Futhermore we take advantage from the improved static type checking at compile time

Reply:
Disagree. 
The data is not coming from a third party, we've full control on it, so there's no need to validate.

I really don't see an user filling the URL with a different JSON, and if so?
It seems ok to me if the site doesn't work if he attempts to do that (right now that's what happen, console shows the zod error rather than value is undefined), however if you really want to avoid that just use a simpler check, there's no need for a validation system.

Enough to check if the decompress method (maybe we neeed to check also parse) fails and show a notification message.

Additionally, if the zod version selection logic will be as planned (had no time to work on it yet), we may drop the zod npm package.

Reply:
I want to point out that You don't know the type of the data after the decompression. In fact typescript declares it as 'any' type.
Another potential issue: What if we decide to change the structure of data saved to the url over the time? How do you ensure that data to respect the current structure do you expect?
This is what zod is for.
In case we need remove zod dep we will consider to remove it (since it is used in other parts of application). Honestly I'd like to keep in our codebase

Reply:
>I want to point out that You don't know the type of the data after the decompression.

That's not correct. You've full control on that, unless the user tries to do some weird unsupported stuff in which case it is correct that the website would not work (or either we can show a notification, even in case of decompression/parse error, there's no need to use zod).

>Another potential issue: What if we decide to change the structure of data saved to the url over the time? How do you ensure that data to respect the current structure do you expect?

Not related to zod, you have to deal with the different content and implement a logic accordingly anyway.
Zod won't solve that problem, if the JSON structure changes, the logic that handles it must change accordingly.
Honestly I can't think of any possible breaking change since you can freely see the json content and act accordingly, but eventually zod won't help you either, it would just say in console "hey, that's not the content your application expected" but about that see the point above.

>In fact typescript declares it as 'any' type.

I guess you're referring to the `JSON.parse` method, and that's right since it would work with any json string.
It's up to us to know what's the ouput and handle it, in that specific case, we indeed are using AppData type as return type on the fuction, or alternative you can simply declare `const appData: AppData = JSON.parse(...)`.
There's no need to use zod for that, nor is related to it.

> This is what zod is for.

Not really, it's mainly for validate content which you've no control on, and eventually give a feedback about what's wrong to the user (it's likely the most common usage is in API requests).
Here we do have control of the input, and there's no need to give a properly formatted error message to an user which is trying to do non-sense stuff (and still if your really want to do that, zod is overkill in that case).

>In case we need remove zod dep we will consider to remove it (since it is used in other parts of application). Honestly I'd like to keep in our codebase

The class which would allow version selection should include everything itself, but that's another topic (and I had no time to start implementing it yet).